ui/app/src/views/ViewDashboard.tsx

```diff
@@ -12,43 +12,122 @@
 import { useDatasourceApi } from '../model/datasource-api';
 import DashboardBreadcrumbs from '../components/DashboardBreadcrumbs';
 import { useIsReadonly } from '../model/config-client';
 import { useSnackbar } from '../context/SnackbarProvider';
+import { CreateAction } from '../model/action';
+
+/**
+ * Generated a resource name valid for the API.
+ * By removing accents from alpha characters and replace specials character by underscores.
+ * @param name
+ */
+function generateMetadataName(name: string): string {
+  return name
+    .normalize('NFD')
+    .replace(/\p{Diacritic}/gu, '')
+    .replace(/[^a-zA-Z0-9_.:-]/g, '_');
+}
 
 /**
  * The View for viewing a Dashboard.
  */
 function ViewDashboard() {
-  const { projectName, dashboardName } = useParams();
+  const { projectName, dashboardName, action } = useParams();
+  const actionRef = useRef(action?.toLowerCase());
 
   if (projectName === undefined || dashboardName === undefined) {
-    throw new Error('Unable to get the Dashboard or Project name');
+    throw new Error('Unable to get the dashboard or project name');
   }
 
+  const navigate = useNavigate();
+  const { successSnackbar, exceptionSnackbar, warningSnackbar } = useSnackbar();
   const datasourceApi = useDatasourceApi();
-  const { data, isLoading } = useDashboard(projectName, dashboardName);
-  const { successSnackbar, exceptionSnackbar } = useSnackbar();
-  const dashboardUpdatePromise = (data: DashboardResource) => {
-    return updateDashboard(data)
-      .then((updatedDashboard) => {
-        successSnackbar(`dashboard ${updatedDashboard.metadata.name} is successfully updated`);
-        return updatedDashboard;
-      })
-      .catch((err) => {
-        exceptionSnackbar(err);
-        throw err;
-      });
-  };
+  const { isLoading } = useDashboard(projectName, dashboardName);
+  let { data } = useDashboard(projectName, dashboardName);
+  console.log(data);
   const isReadonly = useIsReadonly();
+
+  const createDashboardMutation = useCreateDashboardMutation();
+  const updateDashboardMutation = useUpdateDashboardMutation();
+
+  let isEditing = false;
+
+  if (actionRef.current === CreateAction) {
+    if (data !== undefined) {
+      // Dashboard already exists in the API, the user is redirected to the existing dashboard
+      actionRef.current = undefined;
+      warningSnackbar(`Dashboard ${dashboardDisplayName(data)} already exists`);
+      navigate(`/projects/${data.metadata.project}/dashboards/${data.metadata.name}`);
+    } else {
+      data = {
+        kind: 'Dashboard',
+        metadata: {
+          name: generateMetadataName(dashboardName),
+          project: projectName,
+          version: 0,
+        },
+        spec: {
+          display: {
+            name: dashboardName,
+          },
+          duration: '5m',
+          variables: [],
+          layouts: [],
+          panels: {},
+        },
+      } as unknown as DashboardResource;
```

Comment:
Is the `as unknown` part needed?

Reply:
Yes, because I didn't set `created_at` and `updated_at` fields in `metadata` => error. It's the API that fill these fields, so I can either do this or put them as optional in [https://github.com/perses/perses/blob/main/ui/core/src/model/resource.ts#L16](https://github.com/perses/perses/blob/main/ui/core/src/model/resource.ts#L16)

Reply:
Any thoughts here @cndonovan about potentially making `created_at` and `updated_at` optional?